core/src/price_estimation/clients/generic_client.rs

```diff
@@ -1,79 +1,111 @@
-mod api;
-
-use super::{PriceSource, TokenData};
+use super::super::{PriceSource, TokenData};
 use crate::http::HttpFactory;
 use crate::models::TokenId;
 use anyhow::{anyhow, Context, Result};
-use api::{DexagApi, DexagHttpApi};
 use futures::{
     future::{self, BoxFuture, FutureExt as _},
     lock::Mutex,
 };
 use std::collections::HashMap;
 
-struct ApiTokens {
+/// Provides a generic interface to communicate in a standardized way
+/// with specific API token implementations
+pub trait GenericToken {
+    /// Symbol describing the ERC20 token represented by the type instance
+    fn symbol(&self) -> &str;
+}
+
+#[cfg(test)]
+#[derive(Clone, Debug, PartialEq)]
+// Cannot autogenerate with Mockall since the derived traits are needed
+// for testing. GenericToken is a trait that is assigned to the internal
+// token representation of the price source, so the output of `.symbol()`
+// isn't expected to change.
+pub struct MockGenericToken(String);
+#[cfg(test)]
+impl GenericToken for MockGenericToken {
+    fn symbol(&self) -> &str {
+        &self.0
+    }
+}
+#[cfg(test)]
+impl From<&str> for MockGenericToken {
+    fn from(name: &str) -> Self {
+        MockGenericToken(name.to_string())
+    }
+}
+
+#[cfg_attr(test, mockall::automock(type Token=MockGenericToken;))]
```

Comment:
You should be able to mock MockGenericToken in the tests module below and reference it here with `tests::MockGenericToken`